extra/3-magic-8-ball.js

```diff
@@ -43,10 +43,48 @@
     Very doubtful.
 */
 
+let randomAnswer = 0;
+
+let veryPositiveAnswers = [
+    `It is certain.`,
+    `It is decidedly so.`,
+    `Without a doubt.`,
+    `Yes - definitely.`,
+    `You may rely on it.`
+]
+
+let positiveAnswers = [
+    `As I see it, yes.`,
+    `Most likely.`,
+    `Outlook good.`,
+    `Yes.`,
+    `Signs point to yes.`
+]
+
+let negativeAnswers = [
+    `Reply hazy, try again.`,
+    `Ask again later.`,
+    `Better not tell you now.`,
+    `Cannot predict now.`,
+    `Concentrate and ask again.`
+]
+
+let veryNegativeAnswers = [
+    `Don't count on it.`,
+    `My reply is no.`,
+    `My sources say no.`,
+    `Outlook not so good.`,
+    `Very doubtful.`
+]
+
+let allAnswers = veryPositiveAnswers.concat(positiveAnswers, negativeAnswers, veryNegativeAnswers);
+
 // This should log "The ball has shaken!"
 // and return the answer.
 function shakeBall() {
-  //Write your code in here
+  console.log(`The ball has shaken!`);
+  randomAnswer = allAnswers[Math.floor(Math.random() * allAnswers.length)];
```

Comment:
Nitpick: it would be better to use `let` to declare the randomAnswer variable here in the function rather than further up in the script.

Reply:
That's good to know, thanks!